goal_src/jak2/levels/common/airlock.gc

```diff
@@ -7,3 +7,1762 @@
+  (set! (-> obj pre-open-frame) 0.0)
+  (set! (-> obj lock-frame) 0.0)
+  (set! (-> obj pre-open-speed) 2.0)
+  (set! (-> obj open-distance) (res-lump-float (-> obj entity) 'distance :default 143360.0))
+  (set! (-> obj active-distance)
+        (res-lump-float (-> obj entity) 'idle-distance :default (+ 143360.0 (-> obj open-distance)))
+        )
+  (set! (-> obj y-height) (res-lump-data (-> obj entity) 'height vector))
+  (set! (-> obj level-name) (res-lump-struct (-> obj entity) 'on-notice string))
+  (set! (-> obj open-test)
+        (the-as symbol ((method-of-type res-lump get-property-struct)
+                        (-> obj entity)
+                        'open-test
+                        'interp
+                        -1000000000.0
+                        (the-as structure '(not (or (scene-player?) (focus-test? *target* grabbed))))
+                        (the-as (pointer res-tag) #f)
+                        *res-static-buf*
+                        )
+                )
+        )
+  (set! (-> obj sound-gear) #f)
+  (set! (-> obj sound-pre-open) #f)
+  (set! (-> obj sound-pre-open-stop) #f)
+  (set! (-> obj sound-lock-loop) #f)
+  (set! (-> obj sound-lock-stop) #f)
+  (set! (-> obj sound-post-close) #f)
+  (set! (-> obj sound-post-close-stop) #f)
+  (set! (-> obj sound-open) #f)
+  (set! (-> obj sound-close) #f)
+  (set! (-> obj sound-open-loop) #f)
+  (set! (-> obj sound-close-loop) #f)
+  (set! (-> obj sound-open-stop) #f)
+  (set! (-> obj sound-close-stop) #f)
+  (set! (-> obj door-radius) 20480.0)
+  obj
+  )
+
+(defmethod com-airlock-method-25 com-airlock ((obj com-airlock) (arg0 vector) (arg1 symbol))
+  (let ((s5-0 (vector-z-quaternion! (new 'stack-no-clear 'vector) (-> obj root-override quat)))
+        (s4-1 (vector-! (new 'stack-no-clear 'vector) arg0 (-> obj root-override trans)))
+        )
+    (set! (-> s4-1 y) 0.0)
+    (let ((f30-0 (vector-dot s4-1 s5-0)))
+      (cond
+        ((not arg1)
+         )
+        ((or (< (vector-vector-xz-distance (-> obj root-override trans) arg0) 40960.0)
+             (< 0.7 (fabs (vector-dot s5-0 (vector-normalize! s4-1 1.0))))
+             )
+         (when (and (< f30-0 0.0) (< 0.0 (-> obj last-distance)))
+           (let ((s5-1 (res-lump-struct (-> obj entity) 'on-cross structure)))
+             (if s5-1
+                 (script-context-method-9
+                   (new 'stack 'script-context (the-as basic (process->ppointer self)) self (the-as vector #f))
+                   s5-1
+                   )
+                 )
+             )
+           )
+         (set! (-> obj last-distance) f30-0)
+         )
+        ((< 0.0 (-> obj last-distance))
+         (set! f30-0 (fmax 4096.0 f30-0))
+         )
+        ((< (-> obj last-distance) 0.0)
+         (set! f30-0 (fmin -4096.0 f30-0))
+         )
+        )
+      f30-0
+      )
+    )
+  )
+
+;; WARN: Return type mismatch object vs symbol.
+(defmethod open-door? com-airlock ((obj com-airlock))
```

Comment:
I don't this actually checks if we should open the door, but instead decides if we want to go into the airlock or out. Maybe `want-cross-airlock?`